src/components/Toolbar.vue

```diff
@@ -120,13 +127,22 @@ export default {
     redo() {
       this.$store.timetravel.redo();
     },
+    updateUnits() {
+      if (this.allowSettingUnits) {
+        this.rwUnits = this.$refs.unitSelect.value;
+      } else {
+        window.eventBus.$emit('error', 'Units must be set before any geometry is drawn.');
+        this.$refs.unitSelect.value = this.rwUnits;
+      }
+    },
   },
   computed: {
     ...mapState({
       currentMode: state => state.application.currentSelections.mode,
       mapEnabled: state => state.project.map.enabled,
       timetravelInitialized: state => state.timetravelInitialized,
       showImportExport: state => state.project.showImportExport,
+      allowSettingUnits: state => state.geometry.length === 1 && state.geometry[0].vertices.length === 0,
```

Comment:
if they destroy all geometry, should they be able to change the units? This might open up cases where other objects with values in units are stored incorrectly